handler/handler.go

```diff
@@ -0,0 +1,267 @@
+package handler
+
+import (
+	"encoding/json"
+	"go_final_project/function"
+	"go_final_project/structurs"
+	"net/http"
+	"strconv"
+	"time"
+)
+
+type Handler struct {
+	Repo HandlerRepository
+}
+type HandlerRepository interface {
+	AddDB(DBFile string) error
+	CheckTable(DBFile string) error
+	CreateTable(DBFile string) error
+	AddTask(t structurs.Task) (int, error)
+	GetTasks() ([]structurs.Tasks, error)
+	GetTaskId(id string) (structurs.Tasks, error)
+	SearchTask(input string) ([]structurs.Tasks, error)
+	PutTaskId(t structurs.Tasks) error
+	DeleteTaskId(id string) error
+	NextDate(now string, data string, repeat string) (string, error)
+}
+
+func NewHandler(repo HandlerRepository) Handler {
+	return Handler{repo}
+}
+func (h Handler) PostGetPutDeleteTask(w http.ResponseWriter, r *http.Request) {
+	if r.Method == http.MethodPost {
+		h.PostTask(w, r)
+	} else if r.Method == http.MethodGet {
+		h.GetTaskId(w, r)
+	} else if r.Method == http.MethodDelete {
+		h.DeleteTaskID(w, r)
+	} else if r.Method == http.MethodPut {
+		h.PutTask(w, r)
+	}
+
+}
+func (h Handler) PostTask(w http.ResponseWriter, r *http.Request) {
+	resp := make(map[string]string)
+	var TaskAdd structurs.Task
+	w.Header().Set("Content-Type", "application/json; charset=UTF-8")
+	err := json.NewDecoder(r.Body).Decode(&TaskAdd)
+	if err != nil {
+		resp["error"] = err.Error()
+		json.NewEncoder(w).Encode(resp)
+		w.Header().Set("Content-Type", "application/json")
+		return
+	}
+	date, _ := function.DataCheck(TaskAdd.Date)
+	repeat, _ := function.RepeatChek(TaskAdd.Repeat)
+	TaskAdd.Repeat = repeat
+	TaskAdd.Date = date
+	if TaskAdd.Title != "" && TaskAdd.Date != "" && TaskAdd.Repeat != "0" {
```

Comment:
Можно инвертировать условие, ответить клиенту ошибкой и выйти из функции, тогда все что сейчас в IF можно вынести дальше на уровень выше
```suggestion
	if !(TaskAdd.Title != "" && TaskAdd.Date != "" && TaskAdd.Repeat != "0") {
```

Reply:
не совсем понял можно подробней